src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php

```diff
@@ -268,30 +267,30 @@ protected function assertValidMappingConfiguration(array $mappingConfig, $object
      */
     protected function detectMetadataDriver($dir, ContainerBuilder $container)
     {
-        // add the closest existing directory as a resource
         $configPath = $this->getMappingResourceConfigDirectory();
-        $resource = $dir.'/'.$configPath;
-        while (!is_dir($resource)) {
-            $resource = dirname($resource);
-        }
+        $extension = $this->getMappingResourceExtension();
 
-        $container->addResource(new FileResource($resource));
+        if (glob($dir.'/'.$configPath.'/*.'.$extension.'.xml')) {
+            $driver = 'xml';
+        } elseif (glob($dir.'/'.$configPath.'/*.'.$extension.'.yml')) {
+            $driver = 'yml';
+        } elseif (glob($dir.'/'.$configPath.'/*.'.$extension.'.php')) {
+            $driver = 'php';
+        } else {
+            // add the directory itself as a resource
+            $container->fileExists($dir);
```

Comment:
I would also rather add ``$dir.'/'.$configPath`` as a resource, as creating the directory and adding files in it could change the config (adding the whole bundle as a resource when ``<bundle>/Resources`` does not exist was a mistake IMO, hurting performance for some people using annotation, and not working fine anyway as it was not detecting the creation of the folder which would change the type)